internal/agent/device/applications/podman_monitor.go

```diff
@@ -695,10 +728,20 @@ func (m *PodmanMonitor) updateQuadletContainerStatus(ctx context.Context, app Ap
 	if isFinishedStatus(status) && event.ContainerExitCode == 0 {
 		status = StatusExited
 	}
-	m.updateApplicationStatus(app, event, status, restartCount)
+
+	if status == StatusRemove {
```

Comment:
Quadlets are different in annoying ways. This is one of them. Containers don't live outside of the lifecycle of a single start/stop of the systemd unit that controls them. `systemctl stop <quadlet.service>` will generate a removal event, but we don't necessarily want to lose that workload. In this case, I purposely left out removing workloads.